### ISSUE TITLE
Adding a compiler should not remove defaults

### ISSUE THREAD

**Issue by zyklus** [wallabyjs/public#261]
According to the docs, in order to support babel you need to add a new compiler.  What the docs don't say is that you also need to re-add the default compilers (e.g. for coffeescript) in order for them to still work.

Suggestion: Adding new compilers should only affect their matching blobs.  Since I don't add a compiler that matches `foo.coffee`, the internal `**/*.coffee` should still work

**Comment by ArtemGovorov** [comment]
Docs actually [do say](http://wallabyjs.com/docs/config/compilers.html) (paragraph 4) that you need to specify all compilers you're using explicitly if you're changing the `compilers` setting.

How would you suggest to determine whether to use or not use the internal `**/*.coffee` compiler in case if user defined compilers setting was supplied? What would the rules be?

**Comment by zyklus** [comment]
Okay, I was looking at the options for [es6](http://wallabyjs.com/docs/integration/es-next.html).  It's very hard to follow the docs to be honest.

> How would you suggest to determine whether to use or not use the internal **/*.coffee compiler in case if user defined compilers setting was supplied?

```js
for(file in tests) {
  if(any user compiler glob matches file){
    userCompiler()
  } else {
    defaultCompiler()
  }
}
```

**Comment by ArtemGovorov** [comment]
> It's very hard to follow the docs to be honest.

Thanks for the honest feedback. What can we do to improve it? Would adding a note similar to what we have in the compilers section to the es6 section be helpful? Is there anything else you've spotted in the docs that we could improve?

I don't see any reasons why the proposed solution may not work. Will go ahead and implement it.

**Comment by ArtemGovorov** [comment]
I have been thinking it over further, and it looks like there's at least one case that's going to be broken if we always try to use default compilers.

If one has a postprocessor (for example for [Webpack](http://wallabyjs.com/docs/integration/webpack.html) or [Browserify](http://wallabyjs.com/docs/integration/browserify.html)), uses for example CoffeeScript, and wants to compile it with a module bundler plugin (such as [coffeeify](https://github.com/jnordberg/coffeeify) or [coffee-loader](https://github.com/webpack/coffee-loader)), then now one would just set `compilers: {}` to make wallaby not apply the default compiler. Always using internal array of compilers will break the scenario.

**Comment by ArtemGovorov** [comment]
Actually disregard my deleted comment about the breaking change, apparently it's not the case.

**Comment by ArtemGovorov** [comment]
The suggestion is implemented in core v1.0.105. I have also updated the docs to reflect the change.

Please let me know if you have any feedback/suggestions on how I could improve the docs.

**Comment by ArtemGovorov** [comment]
@zyklus Very nice job on writing the wallaby.js in Atom article by the way! I have fixed the issue with the tree view context menu config selection, so should be working now. Please let me know of any other issues you've with wallaby, more than happy to fix them.
